src/dfcx_scrapi/core/security_settings.py

```diff
@@ -0,0 +1,215 @@
+
+import json
+import logging
+from google.cloud.dialogflowcx_v3beta1 import services
+from google.cloud.dialogflowcx_v3beta1 import types
+from google.protobuf import field_mask_pb2
+from dfcx_scrapi.core.scrapi_base import ScrapiBase
+
+
+# logging config
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+class SecuritySettings(ScrapiBase):
+    """Core Class for CCAI Security Settings."""
+
+    def __init__(
+        self,
+        creds_path: str = None,
+        creds_dict: dict = None,
+        creds=None,
+        scope=False,
+        agent_id: str = None
+    ):
+        super().__init__(
+            creds_path=creds_path,
+            creds_dict=creds_dict,
+            creds=creds,
+            scope=scope,
+        )
+
+        if agent_id:
+            self.agent_id = agent_id
+
+        self.ss_service = services.security_settings_service
+        self.ss_types = types.security_settings
+
+    def list_security_settings(self, location_id: str):
+        """List Security Settings for a given Project and Region.
+
+        Args:
+            location_id: CX Agent ID string in the following format
+            projects/<PROJECT ID>/locations/<LOCATION ID>/
+
+        Returns:
+            results: Dict of results
+        """
+
+        request = self.ss_types.ListSecuritySettingsRequest()
+
+        request.parent = location_id
+        client_options = self._set_region(location_id)
+        client = self.ss_service.SecuritySettingsServiceClient(
+            credentials=self.creds, client_options=client_options
+        )
+
+        response = client.list_security_settings(request)
+
+        security_settings = []
+        for page in response.pages:
+            for ss in page.security_settings:
+                security_settings.append(ss)
+
+        return security_settings
+
+
+    def get_security_settings(self, security_setting_id: str):
+        """Get specified CCAI Security Setting.
+
+        Args:
+            security_setting_id: Security Setting ID string in the following
+                format: projects/<PROJECT ID>/locations/<LOCATION ID>/
+                securitySettings/<SECURITY SETTINGS ID>
+
+        Returns:
+            response: a single Security Settings object of
+              types.SecuritySettings
+        """
+
+        request = self.ss_types.GetSecuritySettingsRequest()
+
+        request.name = security_setting_id
+        client_options = self._set_region(security_setting_id)
+        client = self.ss_service.SecuritySettingsServiceClient(
+            credentials=self.creds, client_options=client_options
+        )
+
+        response = client.get_security_settings(request)
+
+        return response
+
+    def create_security_settings(
+        self,
+        location_id: str,
+        obj: types.SecuritySettings = None,
+        security_settings_dict: dict[str,str] = None):
```

Comment:
capitalize d in dict type hint @kmaphoenix

Reply:
ack.